torch/_torch_docs.py

```diff
@@ -740,11 +740,17 @@ def parse_kwargs(desc):
 
 add_docstr(torch.cat,
            r"""
-cat(seq, dim=0, out=None) -> Tensor
+cat(seq, dim=0, pad_value=None, out=None) -> Tensor
 
 Concatenates the given sequence of :attr:`seq` tensors in the given dimension.
-All tensors must either have the same shape (except in the concatenating
-dimension) or be empty.
+If pad_value is set to None, all tensors must have the same shape,
+except in the dimension corresponding to :attr:`dim`.
+If pad_value is set to a scalar, all tensors only need to have the same number
+of dimensions.The size of the resulting tensor is the size of the sum of the
+dimensions in the concatenating dimension, and the maximum of the
+dimension sizes in all other dimensions. If an input tensor needs to be
```

Comment:
nit: it is probably nicer to give a mathematical example:
we're cat-ing tensors with size (x0, x1, ..., xn), (y0, y1, ..., yn)...